recipes/bitserializer/all/conanfile.py

```diff
@@ -45,21 +50,33 @@ def _compilers_minimum_version(self):
             "apple-clang": "12",
         }
 
+    def _patch_sources(self):
+        # Remove 'ryml' subdirectory from #include
+        replace_in_file(self, os.path.join(self.source_folder, "include/bitserializer/rapidyaml_archive.h"), "#include <ryml/", "#include <")
+
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_cpprestsdk:
             self.requires("cpprestsdk/2.10.18", transitive_headers=True, transitive_libs=True)
         if self.options.with_rapidjson:
-            self.requires("rapidjson/cci.20220514", transitive_headers=True, transitive_libs=True)
+            self.requires("rapidjson/cci.20220822", transitive_headers=True, transitive_libs=True)
         if self.options.with_pugixml:
             self.requires("pugixml/1.13", transitive_headers=True, transitive_libs=True)
-
-    def package_id(self):
-        self.info.clear()
+        if self.options.with_rapidyaml:
+            self.requires("rapidyaml/0.4.1", transitive_headers=True, transitive_libs=True)
```

Comment:
```suggestion
            self.requires("rapidyaml/0.4.1")
```
Rapidyaml is not header-only. Transitive headers is to be used for those cases where a public header consumes another dependency